mitodl/ocw-hugo-themes
# Fix table code erroring on non-table pages

#### Pre-Flight checklist


#### What are the relevant tickets?
#875 followup to #844 

#### What's this PR do?
This PR:
- fixes an issue with `responsive_tables.js` where code was assuming that every page had at least one table.
- fixes an issue where only the first table on each page was respnsive
- converts course_info_toggle.js and responsive_tables.js to typescript

#### How should this be manually tested?
First, visit any page on ocw.mit.edu that does *not* have a table. Observe the console errors.

Now on this branch using the original course theme:
1. Visit any page without a table. Check that there are no errors.
2. Now check a page with large (wide) tables...
3. Open the content markdown for any page. Replace it with https://gist.github.com/ChristopherChudzicki/dd0d177987b2baf219dfb9b2f351a097
4. View the hugo build for that page. Check that clicking "Show course info / hide course info" expands/collapses the tables appropriately.

#### Screenshots (if appropriate)

https://user-images.githubusercontent.com/9010790/193902339-fc545751-f681-49e5-a4d8-8087483961bb.mov
